crates/tools/trust_db_util.rs

```diff
@@ -0,0 +1,352 @@
+use std::fs::File;
+use std::io;
+use std::io::{BufReader, Write};
+use std::path::Path;
+use std::process::{Command, Output};
+use std::time::SystemTime;
+
+use clap::Clap;
+use lmdb::{DatabaseFlags, Environment, Transaction, WriteFlags};
+use rayon::prelude::*;
+use thiserror::Error;
+
+use fapolicy_api::trust::Trust;
+use fapolicy_app::cfg;
+use fapolicy_daemon::rpm::load_system_trust as load_rpm_trust;
+use fapolicy_trust::read;
+use fapolicy_trust::read::check_trust_db;
+use fapolicy_util::sha::sha256_digest;
+
+use crate::Error::{DirTrustError, DpkgCommandFail, DpkgNotFound};
+use crate::Subcommand::{Add, Check, Clear, Del, Dump, Init, Search};
+use fapolicy_daemon::fapolicyd::TRUST_DB_NAME;
+
+/// An Error that can occur in this app
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("dpkg-query not found on path")]
+    DpkgNotFound,
+
+    #[error("dpkg-query command failed, {0}")]
+    DpkgCommandFail(io::Error),
+
+    #[error("{0}")]
+    RpmError(#[from] fapolicy_daemon::rpm::Error),
+
+    #[error("{0}")]
+    LmdbError(#[from] lmdb::Error),
+
+    #[error("{0}")]
+    TrustError(#[from] fapolicy_trust::error::Error),
+
+    #[error("cant trust a directory")]
+    DirTrustError,
+
+    #[error("file io error, {0}")]
+    FileError(#[from] io::Error),
+
+    #[error("file hashing error, {0}")]
+    HashError(#[from] fapolicy_util::error::Error),
+
+    #[error("error reading stdout")]
+    ParseError(#[from] std::string::FromUtf8Error),
+}
+
+#[derive(Clap)]
+#[clap(name = "Trust DB Utils", version = "v0.1")]
+struct Opts {
+    #[clap(subcommand)]
+    pub cmd: Subcommand,
+
+    /// Specify the fapolicyd database directory
+    /// Defaults to the value specified in the users XDG configuration
+    dbdir: Option<String>,
+}
+
+#[derive(Clap)]
+enum Subcommand {
+    /// Clear the trust database of all entries
+    Clear(ClearOpts),
+    /// Initialize the trust database from system trust
+    Init(InitOpts),
+    /// Add a file to the trust database
+    Add(AddRecOpts),
+    /// Remove a file from the trust database
+    Del(DelRecOpts),
+    /// Dump all trust entries to file or stdout
+    Dump(DumpDbOpts),
+    /// Search for a trust entry
+    Search(SearchDbOpts),
+    /// Check trust status
+    Check(CheckDbOpts),
+}
+
+#[derive(Clap)]
+struct ClearOpts {}
+
+#[derive(Clap)]
+struct InitOpts {
+    /// create an empty database
+    #[clap(long)]
+    empty: bool,
+
+    /// overwrite existing database
+    #[clap(long)]
+    force: bool,
+
+    /// use dpkg to generate db
+    #[clap(long)]
+    dpkg: bool,
```

Comment:
If there were more than two possible states right now it would have been.  With an enum you will still have RPM as the default and pass a flag to override to dpkg, same as it is now.